src/providers/core/mod.rs

```diff
@@ -326,4 +340,17 @@ mod tests {
             provider.wire_protocol_version_min
         );
     }
+
+    #[test]
+    fn test_build() {
+        let provider_builder = ProviderBuilder::new().with_wire_protocol_version(42, 12);
+        let provider = provider_builder.build().unwrap();
+        for provider_info in provider.provider_info {
+            assert!(
```

Comment:
I can't find anything to say that version 0.0.0 is illegal (even if it is unlikely) so this might not be a valid check. I'm not sure we need to do anything beyond checking the version can be parsed into 3 integers.

Reply:
Would it be sufficient to call `provider_builder.build().unwrap()` or did you have any specific assertions in mind?

Reply:
Just check the format should be sufficient enough I guess.